File: output/ChatGPT/avatar/Python/Java/codeforces_67_B.java
int n = input.nextInt();
int k = input.nextInt();
int[] M = new int[n];
for(int i=0; i<n; i++){
    M[i] = input.nextInt();
}
int m = 0;
for(int j=0; j<n; j++){
    int p = 0;
    for(p=0; p<n; p++){
        if(M[p] == 0){
            System.out.print(p+1 + " ");
            break;
        }
    }
    for(int l=0; l<p+1-k; l++){
        M[l] -= 1;
    }
    M[p] -= 1;
}

File: output/ChatGPT/avatar/Java/Python/codeforces_67_B.py
n, k = map(int, input().split())
bb = list(map(int, input().split()))
aa = [0] * n
m = 0
for a in range(n - 1, -1, -1):
    j = 0
    while bb[a] > 0:
        if aa[j] >= a + k:
            bb[a] -= 1
        j += 1
    for j_ in range(m, j, -1):
        aa[j_] = aa[j_ - 1]
    aa[j] = a
for i in range(n):
    print(aa[i] + 1, end=" ")